panther_lights/src/controller_node.py

```diff
@@ -21,10 +20,77 @@
 
 
 class PantherAnimation:
+    ANIMATION_DEFAULT_PRIORITY = 3
+    ANIMATION_DEFAULT_TIMEOUT = 120
+
     front: Animation
     rear: Animation
-    interrupting = False
-    repeating = False
+    name: str = 'UNDEFINED'
+    priority: int = ANIMATION_DEFAULT_PRIORITY
+    timeout: float = ANIMATION_DEFAULT_TIMEOUT
+    repeating: bool = False
+
+    def __init__(self) -> None:
+        self._init_time = rospy.get_time()
+
+    def reset_time(self) -> None:
+        self._init_time = rospy.get_time()
+
+    @property
+    def init_time(self) -> float:
+        return self._init_time
+
+
+class AnimationsQueue:
+    def __init__(self, max_queue_size) -> None:
+        self._queue = []
+        self._max_queue_size = max_queue_size
+
+    def put(self, animation: PantherAnimation) -> None:
+        if animation.priority == 1:
+            self.clear()
+        self.validate_queue()
+
+        if len(self._queue) == self._max_queue_size:
+            rospy.logwarn(f'{rospy.get_name()} Animation queue overloaded')
+            return
+
+        self._queue.append(animation)
+        self._queue = sorted(self._queue, key=lambda x: (x.priority, x.init_time))
+
+    def get(self) -> PantherAnimation:
+        return self._queue.pop(0)
```

Comment:
```suggestion
    def pop(self) -> PantherAnimation:
        return self._queue.pop(0)
```
I think `pop` might more descriptive since it will tell exactly what it does under the hood

Reply:
I wanted to make custom queue to be similar to standard python `Queue` which has get method. I think it makes more sense and is more intuitive for people who used standard queue before.

Reply:
Ok. Makes sense